embassy-rp/src/dma.rs

```diff
@@ -174,6 +174,19 @@ impl<'a, C: Channel> Transfer<'a, C> {
 
         Self { channel }
     }
+    /// Abort a DMA transfer early
+    ///
+    /// Returns the count of transfers still left to do
+    pub fn abort(self) -> usize {
+        let p = self.channel.regs();
+        let transfer_count = p.trans_count().read();
```

Comment:
maybe you need to read trans_count after stopping? I think it's possible for some more transfers to go through between reading this and stopping?

Reply:
No, afaik after the abort the transfer count will be reset to zero

Reply:
hmm i'm not sure how to do this then, but I'm pretty sure the current code has a race condition.